Snakemake/workflow/rules/merge_samples_vcf.smk

```diff
@@ -14,34 +14,61 @@ if len(allFiles) != 1:
             """
             bcftools query -l {input} > {output}
             """
-
-    rule compare:
-        input:
-            samples = [f'{vcfdir}' + x
+    
+    rule filter:
+        input: 
+            [f'{vcfdir}' + x
                 for x in expand('/{{chromosome}}/{{chromosome}}_{file}.txt',
                     file=allFiles)],
-            vcfs = [f'{vcfdir}' + x
-                for x in expand('/{{chromosome}}/{{chromosome}}_{file}.vcf.gz',
-                    file=allFiles)]
-        output:
+        output: 
             temp([f'{vcfdir}' + x for x in
-                expand('/{{chromosome}}/{{chromosome}}_{file}.filtered.vcf.gz',
-                    file=allFiles)])
-#        log: 'logs/compare_{chromosome}_{file}.log'
-        conda: "bcftools"
-        threads: 1
-        resources: cpus=1, mem_mb=4000, time_min=5
-        shell:
-            """
-            python scripts/CompareVCFs.py {input.samples}  {input.vcfs}  {output}
-            """
+                expand('/{{chromosome}}/{{chromosome}}_{file}.filtered.vcf.{ext}', file=allFiles, ext=['gz', 'gz.csi'])])
+        params:
+            duplicated = '{chromosome}.ids'
+       # conda: 'bcftools'
+        run:
+            import os
+            filtered = []
+
+            for i in range(len(input)):    
+                print(i)
+                for j, file in enumerate(input):
+                    if file != input[i] and j > i:
+                        file1 = input[i]
+                        file2 = input[j]
+                        print(file1, file2)
+
+                        shell("comm -12 <(sort {file1}) <(sort {file2}) > {params.duplicated}")
+
+                        remove = params.duplicated
+                        lremove = os.stat(f'{remove}').st_size
+                       
+                        if lremove == 0:
+                            print(f'Nothing to remove {lremove}')
+                        else:
+                            print(f'{lremove} samples to remove')
+                            if file2 not in filtered:
+                                filtered.append(file2)
+
+                            file2 = file2.split('.')[0]
+                            print(f'file {file2} filtered')
+                            shell('bcftools view -S ^{params.duplicated} {file2}.vcf.gz -O z -o {file2}.filtered.vcf.gz') 
+                            shell('bcftools index {file2}.filtered.vcf.gz')
+                            shell('rm {params.duplicated}')
+                                           
+            for file in input:
+                if file not in filtered:
+                    file1 = file.split('.')[0]
+                    print(f'file {file1} not filtered')
+                    shell("cp {file1}.vcf.gz {file1}.filtered.vcf.gz && bcftools index {file1}.filtered.vcf.gz")
+            shell('rm {params.duplicated}') 
```

Comment:
What is this bit?

Reply:
This is just to rename files that haven't been filtered - so they all have the same standard name

Reply:
is this the "same" rule as before, just different name of the rule?